### PR TITLE
Allow returning a custom value direcly when handling inputs

### PR DESCRIPTION
**Pull request by bugadani**
Closes #25

### CODE REVIEW

**Comment by bugadani** [comment]
@hansl 

There is definitely some tiny difference in complexity between our solutions, but I wasn't even certain my idea was possible. If you could give this a go to see if I managed to actually cover what you need, that would be awesome!

**Comment by hansl** [comment]
I'll look into this tomorrow. Thanks a lot for the quick response!

**Comment by bugadani** [comment]
I managed to undo some of the changes, so it's probably a bit cleaner now :)

File: src/interaction/mod.rs [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
 pub mod single_touch;
 
 #[cfg(feature = "simulator")]
 pub mod simulator;
 
 #[derive(Copy, Clone, Debug, PartialEq, Eq)]
-pub enum InteractionType {
+pub enum Interaction<R> {
+    /// Equivalent to `BackwardWrapping(1)`, kept for backward compatibility.
+    Navigation(Navigation),
```
Comment (hansl) [inline src/interaction/mod.rs:10]: This is very nice :+1:

File: src/interaction/mod.rs [hunk 2]
```diff
@@ -1,13 +1,27 @@
...
 pub mod programmed;
 pub mod single_touch;
 
 #[cfg(feature = "simulator")]
 pub mod simulator;
 
 #[derive(Copy, Clone, Debug, PartialEq, Eq)]
-pub enum InteractionType {
+pub enum Interaction<R> {
+    /// Equivalent to `BackwardWrapping(1)`, kept for backward compatibility.
```
Comment (hansl) [inline src/interaction/mod.rs]: Comment is invalid.

File: src/interaction/mod.rs (hunk 1 above)
Comment (bugadani) [reply src/interaction/mod.rs:10]: Except for the comment, lol

**Review by hansl — APPROVED** [review]
This is great!

**Comment by bugadani** [comment]
Thanks for the review, much appreciated :)

**Pull request merged**